frameworks/Rust/ntex/src/db.rs

```diff
@@ -161,31 +175,49 @@ impl PgConnection {
         }
     }
 
-    pub fn tell_fortune(&self) -> impl Future<Output = Result<Vec<Fortune>, io::Error>> {
-        let mut items = vec![Fortune {
-            id: 0,
-            message: "Additional fortune added at request time.".to_string(),
-        }];
-
+    pub fn tell_fortune(&self) -> impl Future<Output = Result<Bytes, io::Error>> {
         let fut = self.cl.query_raw(&self.fortune, &[]);
 
         async move {
             let mut stream = fut
                 .await
                 .map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{:?}", e)))?;
 
+            let mut rows = SmallVec::<[_; 32]>::new();
```

Comment:
The changes here and setting the body length capacity violate rule 5. in the fortunes test:

> The list data structure must be a dynamic-size or equivalent and should not be dimensioned using foreknowledge of the row-count of the database table.

Reply:
SmallVec is dynamic, it can contain any number of elements

Reply:
I don't see any documentation on the `TemplateFixed` macro, but it makes me think that using that with these initial capacities is optimizing for a known size of memory allocation. Can you help me understand what `TemplateFixed` does? Also,

```Rust
let mut body = BytesMut::with_capacity(2048);
            unsafe {
                // Maybe uninit
                body.set_len(2048);
```

Reply:
this is just uninitialized buffer, yarte engine write to that memory directly.

https://docs.rs/yarte/0.9.9/yarte/derive.TemplateFixed.html

Reply:
If the fortune body is more than 2048 bytes:
```
ntex-db: thread 'ntex-rt:worker:9' panicked at 'called `Option::unwrap()` on a `None` value', src/db.rs:211:28
```
This can be replicated by adding a couple more rows to the Fortunes table.

Reply:
fair point. it is fixed.

Reply:
@fafhrd91 thanks!